board/app/controllers/user_controller.php

```diff
@@ -31,62 +32,160 @@ public function register()
                 } catch (ValidationException $e) {
                     $page = 'register';
                 }
-
                 break;
             default:
                 throw new NotFoundException("{$page} is not found");
-            break;
+                break;
         }
         $this->set(get_defined_vars());
         $this->render($page);
     }
 
-    public function login()
+    public function login() 
     {
-        if (is_logged_in()) 
-        {
-            redirect(url('user/login_end'));
+        if (is_logged_in()) {
+            redirect(url('user/home'));
         }
 
         $params = array(
             'username' => trim(Param::get('username')),
-            'password' => Param::get('password'), 
+            'password' => Param::get('password')
         );
 
         $user = new User($params);
         $page = Param::get('page_next', 'login');
 
-            switch ($page) {
-                case 'login':
-                break;
+        switch ($page) {
+            case 'login':
+            break;
          
-                case 'login_end':
-                    try 
-                    {
-                        $user->login();
-                    }catch (ValidationException $e){
-                        $page = 'login';
-                    }
-            
-                    break;
-                default:
-                    throw new NotFoundException("{$page} is not found");
-                    break;
-            }
+            case 'home':
+                try {
+                    $user->login();
+                }catch (ValidationException $e){
+                    $page = 'login';
```

Comment:
Done po :bow:
Applied it in all controllers as well. Thank you po. This is actually new to me :blush:
--> https://github.com/maragatchalian/FoodFacts/commit/cef8563369ebbc50573fbda39f3340943a801096